android/src/main/java/com/swmansion/rnscreens/ScreenStackFragment.kt

```diff
@@ -93,48 +126,292 @@ class ScreenStackFragment : ScreenFragment, ScreenStackFragmentWrapper {
         }
     }
 
-    override fun onStart() {
-        lastFocusedChild?.requestFocus()
-        super.onStart()
+    private val bottomSheetStateCallback = object : BottomSheetCallback() {
+        private var lastStableState: Int = Screen.sheetStateFromDetentIndex(screen.sheetInitialDetentIndex, screen.sheetDetents.count())
+
+        override fun onStateChanged(bottomSheet: View, newState: Int) {
+            if (Screen.isStateStable(newState)) {
+                lastStableState = newState
+                screen.emitOnSheetDetentChanged(Screen.detentIndexFromSheetState(lastStableState, screen.sheetDetents.count()), true)
+            } else if (newState == BottomSheetBehavior.STATE_DRAGGING) {
+                screen.emitOnSheetDetentChanged(Screen.detentIndexFromSheetState(lastStableState, screen.sheetDetents.count()), false)
+            }
+            if (newState == BottomSheetBehavior.STATE_HIDDEN) {
+                if (this@ScreenStackFragment.parentFragment is DimmingFragment) {
+                    parentFragmentManager.commit {
+                        setReorderingAllowed(true)
+                        remove(this@ScreenStackFragment)
+                    }
+                } else {
+                    this@ScreenStackFragment.dismissFromContainer()
+                }
+            }
+        }
+
+        override fun onSlide(bottomSheet: View, slideOffset: Float) = Unit
+    }
+
+    override fun onCreateAnimation(transit: Int, enter: Boolean, nextAnim: Int): Animation? {
+        if (screen.stackPresentation != Screen.StackPresentation.FORM_SHEET) {
+            return null
+        }
```

Comment:
`null` returned here is intentional. If something is returned from `onCreateAnimation` it will override the custom transition set during transaction creation - and we do not want this to happen for different types of stack presentation.